www/js/common/map.js

```diff
@@ -44,17 +47,79 @@ angular.module('emission.main.common.map',['ionic-datepicker',
   $scope.refreshMap = function() {
       CommonGraph.updateCurrent();
   };
-
+  $scope.getFormattedDuration = DiaryHelper.getFormattedDuration;
   $scope.$on(CommonGraph.UPDATE_DONE, function(event, args) {
     $scope.$apply(function() {
         $scope.mapCtrl.geojson = {}
         $scope.mapCtrl.geojson.data = CommonGraph.data.geojson;
         $scope.mapCtrl.geojson.style = styleFeature;
         $scope.mapCtrl.geojson.onEachFeature = onEachFeature;
+        $scope.mapCtrl.trips = CommonGraph.data.graph.common_trips;
         // $scope.mapCtrl.geojson.pointToLayer = pointFormat;
     });
   });
 
+
+  /*
+   * if given group is the selected group, deselect it
+   * else, select the given group
+   */
+  $scope.toggleGroup = function(group) {
+    group.show = !group.show;
+    if (group.show) {
+      var vals = [];
+      var probs = group.probabilites[group.start_times[0].weekday];
```

Comment:
I am not sure what you are doing here, but it seems incorrect - it looks like you are returning the probability of the first start time. I see no evidence that the start times are sorted by most popular -> least popular. You need to use `binArray` to get the distribution of start times.

Unless there is an nvd3 `histogram` that will do the computation for you? I think that there's at least a raw d3 `histogram`.
https://github.com/d3/d3-array/blob/master/README.md#histograms